compiler/src/parsing/ast_helper.re

```diff
@@ -319,6 +319,31 @@ module Expression = {
     mk(~loc, ~core_loc, ~attributes?, PExpLambda(a, b));
   let apply = (~loc, ~core_loc, ~attributes=?, a, b) =>
     mk(~loc, ~core_loc, ~attributes?, PExpApp(a, b));
+  let partial_apply = (~loc, ~core_loc, ~attributes=?, a, b) =>
+    mk(~loc, ~core_loc, ~attributes?, PExpPartial(a, b));
+  let total_apply_args = (~loc, ~core_loc, ~attributes=?, a, b) => {
+    let args =
+      List.map(
+        arg =>
+          {
+            paa_loc: arg.ppaa_loc,
+            paa_expr:
+              switch (arg.ppaa_expr) {
+              | ArgumentGiven(expr) => expr
+              | ArgumentHole(_) =>
+                raise(
+                  SyntaxError(
+                    loc,
```

Comment:
Should we point to the `arg.ppaa_loc`? I feel like this error might be a little annoying as it isn't immediately clear that the `_` hole is specific to partial application, at least that way the position would be telling you hey this argument is the issue specifically, an alteranatie here would be `Hole patterns are only allowed under partial application, prefix the function call with partial.` which is a bit more specific and searchable.